src/openfhe.jl

```diff
@@ -381,3 +385,278 @@ function rotate(sv::SecureVector{<:OpenFHEBackend}, shift; wrap_by)
 
     secure_vector
 end
+
+
+############################################################################################
+# Matrix
+############################################################################################
+
+function init_matrix_rotation!(context::SecureContext{<:OpenFHEBackend}, private_key::PrivateKey,
+                               shifts_2d::Vector{Tuple{Int, Int}}, shape::Tuple{Int, Int})
+    cc = get_crypto_context(context)
+    shifts_1d = []
+    for (i, j) in shifts_2d
+        # minimum required shift
+        i %= shape[1]
+        j %= shape[2] 
+        # appropriate shift for matrix packed in vector
+        shift = []
+        if j == 0
+            if i != 0
+                shift = [i*shape[2]]
+            end
+        else
+            shift = [j+i*shape[2], -sign(j)*(shape[2]-abs(j))+i*shape[2]]
+        end
+        append!(shifts_1d, shift)
+        # additional shift required in case of rotation in i direction
+        if i != 0
+            shift_rest = -sign.(shift) .* (shape[1]*shape[2] .- abs.(shift))
+            append!(shifts_1d, shift_rest)
+        end
+    end
+    OpenFHE.EvalRotateKeyGen(cc, private_key.private_key, -shifts_1d)
+
+    nothing
+end
+
+function PlainMatrix(data::Vector{Float64}, context::SecureContext{<:OpenFHEBackend},
+                     shape::Tuple{Int, Int})
+    cc = get_crypto_context(context)
+    plaintext = OpenFHE.MakeCKKSPackedPlaintext(cc, data)
+    capacity = OpenFHE.GetSlots(plaintext)
+    plain_matrix = PlainMatrix(plaintext, shape, capacity, context)
+
+    plain_matrix
+end
+
+function PlainMatrix(data::Vector{<:Real}, context::SecureContext{<:OpenFHEBackend},
+                     shape::Tuple{Int, Int})
+    PlainMatrix(Vector{Float64}(data), context, shape)
+end
+
+function PlainMatrix(data::Matrix{<:Real}, context::SecureContext{<:OpenFHEBackend})
+    PlainMatrix(Vector{Float64}(vec(transpose(data))), context, size(data))
+end
+
+function Base.show(io::IO, m::PlainMatrix{<:OpenFHEBackend})
+    print(io, collect(m))
+end
+
+function Base.show(io::IO, ::MIME"text/plain", m::PlainMatrix{<:OpenFHEBackend})
+    print(io, m.shape, "-shaped PlainMatrix{OpenFHEBackend}:\n")
+    Base.print_matrix(io, collect(m))
+end
+
+function Base.collect(plain_matrix::PlainMatrix{<:OpenFHEBackend})
+    n = plain_matrix.shape[1]
+    m = plain_matrix.shape[2]
+    vec = OpenFHE.GetRealPackedValue(plain_matrix.data)[1:n*m]
+    Matrix{Float64}(transpose(reshape(vec, (m, n))))
```

Comment:
```suggestion
    data = OpenFHE.GetRealPackedValue(plain_matrix.data)[1:n*m]
    Matrix{Float64}(transpose(reshape(data, (m, n))))
```
`vec` is a Julia function and would be shadowed by this (not illegal, but also not nice)